<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>Toggle-dev by SouthGreenPlatform</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">Toggle-dev</h1>
      <h2 class="project-tagline">Toolbox for generic NGS analyses - dev repository</h2>
      <a href="https://github.com/SouthGreenPlatform/TOGGLE-DEV" class="btn">View on GitHub</a>
      <a href="https://github.com/SouthGreenPlatform/TOGGLE-DEV/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/SouthGreenPlatform/TOGGLE-DEV/tarball/master" class="btn">Download .tar.gz</a>
    </section>

    <section class="main-content">
      <h1>
<a id="toggle--toolbox-for-generic-ngs-analyses" class="anchor" href="#toggle--toolbox-for-generic-ngs-analyses" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>TOGGLE : Toolbox for generic NGS analyses</h1>

<p><img src="docs/images/toggleLogo.png" alt="TOGGLE Logo"></p>

<p>Dear Biologist, have you ever dream of using the whole power of those numerous NGS tools that your bioinformatician colleagues use through this awful list of command line ?</p>

<p>Dear Bioinformatician, have you ever guess how to design really fastly a new NGS pipeline without having to retype again dozens of code lines to readapt your scripts or starting from scratch ?</p>

<p><strong>So, be Happy ! TOGGLE is for you !!</strong></p>

<p>TOGGLE (TOolbox for Generic nGs anaLysEs) is a suite of 19 packages and more than 110 modules able to manage a large set of NGS softwares
and utilities to easily design pipelines able to handle hundreds of samples. Moreover, TOGGLE offers an easy way to manipulate the various
options of the different softwares through the pipelines in using a single basic configuration file, that can be changed for each assay without
having to change the code itself.</p>

<p>Users can also create their own pipeline through an easy and user-friendly approach. The pipelines can start from Fastq (plain or gzipped), SAM, BAM or VCF (plain or gzipped) files, with parallel and global analyses. Samples pipelines are provided for SNP discovery and RNAseq counts.</p>

<p>The system is able to detect parallel/scheduling launching and to manage large amount of samples on large cluster machines.</p>

<h2>
<a id="contributing" class="anchor" href="#contributing" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Contributing</h2>

<ul>
<li>Licencied under CeCill-C (<a href="http://www.cecill.info/licences/Licence_CeCILL-C_V1-en.html">http://www.cecill.info/licences/Licence_CeCILL-C_V1-en.html</a>) and GPLv3</li>
<li>Intellectual property belongs to IRD, CIRAD, ADNid and SouthGreen development platform</li>
<li>Written by Cecile Monat, Christine Tranchant, Ayite Kougbeadjo, Cedric Farcy, Mawusse Agbessi, Enrique Ortega-Abboud, Sébastien Ravel, Julie Orjuela-Bouniol, Souhila Amanzougarene, Gauthier Sarah, Marilyne Summo, and Francois Sabot</li>
<li>Copyright 2014-2016</li>
</ul>

<h2>
<a id="contact" class="anchor" href="#contact" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Contact</h2>

<p>For bug tracking purpose you can use the GitHub or questions about TOGGLE, you can contact the mainteners using the following email addresses:</p>

<ul>
<li><a href="mailto:christine.tranchant@ird.fr">christine.tranchant@ird.fr</a></li>
<li><a href="mailto:francois.sabot@ird.fr">francois.sabot@ird.fr</a></li>
</ul>

<h2>
<a id="citation" class="anchor" href="#citation" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Citation</h2>

<p><strong>TOGGLE: Toolbox for generic NGS analyses</strong>. Cécile Monat, Christine Tranchant-Dubreuil, Ayité Kougbeadjo, Cédric Farcy, Enrique
Ortega-Abboud, Souhila Amanzougarene, Sébastien Ravel, Mawussé Agbessi, Julie Orjuela-Bouniol, Maryline Summo and François Sabot.</p>

<p><a href="http://www.biomedcentral.com/1471-2105/16/374"><em>BMC Bioinformatics</em> 2015, 16:374  doi:10.1186/s12859-015-0795-6</a></p>

<h2>
<a id="installation" class="anchor" href="#installation" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>INSTALLATION</h2>

<p><a href="https://github.com/SouthGreenPlatform/TOGGLE/blob/master/docs/INSTALL.md">Follow the INSTALLATION instructions</a></p>

<h2>
<a id="manual" class="anchor" href="#manual" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>MANUAL</h2>

<p><a href="https://github.com/SouthGreenPlatform/TOGGLE/blob/master/docs/MANUAL.md">You can find a detailed MANUAL here</a></p>

<h2>
<a id="known-issues" class="anchor" href="#known-issues" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>KNOWN ISSUES</h2>

<p><a href="https://github.com/SouthGreenPlatform/TOGGLE-DEV/blob/master/docs/KnownIssues.md">You can find detailed known issues</a></p>

<h2>
<a id="release-notes" class="anchor" href="#release-notes" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Release Notes</h2>

<p><a href="https://github.com/SouthGreenPlatform/TOGGLE/blob/master/docs/ReleaseNotes.md">Current Release Notes</a></p>

<h2>
<a id="requirements" class="anchor" href="#requirements" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>REQUIREMENTS</h2>

<h4>
<a id="perl" class="anchor" href="#perl" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Perl</h4>

<ul>
<li><a href="http://search.cpan.org/%7Edagolden/Capture-Tiny-0.30/lib/Capture/Tiny.pm">Capture::Tiny</a></li>
<li><a href="http://search.cpan.org/%7Edavieira/Data_Translate-0.3/Translate.pm">Data::Translate</a></li>
<li><a href="http://search.cpan.org/%7Esmueller/Data-Dumper-2.154/Dumper.pm">Data::Dumper</a></li>
<li><a href="http://search.cpan.org/%7Emytram/Getopt-ArgParse-1.0.2/lib/Getopt/ArgParse.pm">Getopt::ArgParse</a></li>
<li><a href="http://search.cpan.org/%7Ejkeenan/List-Compare-0.53/lib/List/Compare.pm">List::Compare</a></li>
<li><a href="https://metacpan.org/pod/Switch">Switch</a></li>
<li><a href="http://search.cpan.org/%7Eexodist/Test-Simple-1.001014/lib/Test/More.pm">Test::More</a></li>
<li><a href="http://search.cpan.org/%7Erjbs/Test-Deep-0.119/lib/Test/Deep.pm">Test::Deep</a></li>
</ul>

<h4>
<a id="bioinformatics-software-minimal-version" class="anchor" href="#bioinformatics-software-minimal-version" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Bioinformatics software (minimal version)</h4>

<table border="1" cellpadding="5" cellspacing="1">
<thead>
<tr>
    <th> Category        </th>
    <th> Software        </th>
    <th> Minimal version </th>
    <th> Tools included  </th>
</tr>
</thead>
<tbody>
<tr>
    <td align="center" rowspan="2"> <b>System<b>   </b></b>
</td>
    <td> <a href="https://www.java.com">Java</a>
</td>
    <td> 1.7 non open JDK </td>
    <td></td>
</tr>
<tr>
    <td> <a href="https://www.perl.org/">Perl</a>
</td>
    <td> 5.16 </td>
    <td>  </td>
</tr>
<tr>
    <td align="center" rowspan="3"> <b> Mapping <b> </b></b>
</td>
    <td> <a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie2</a>
</td>
    <td> 2.2.9 </td>
    <td> bowtie2build </td>
</tr>
<tr>
    <td> <a href="http://bio-bwa.sourceforge.net/">BWA</a>
</td>
    <td> 0.7.2 </td>
    <td> bwaAln  bwaSampe  bwaSamse  bwaMem  bwaIndex</td>
</tr>
<tr>
    <td> <a href="https://ccb.jhu.edu/software/tophat/index.shtml">Tophat</a>
</td>
    <td> 2.1.1 </td>
    <td> bowtiebuild  tophat2 </td>
</tr>
<tr>
    <td align="center" rowspan="6"> <b> Fastq/BAM/SAM  tools <b> </b></b>
</td>
    <td> <a href="https://www.broadinstitute.org/gatk/">GATK</a>
</td>
    <td> 3.3 </td>
    <td> gatkBaseRecalibrator  gatkRealignerTargetCreator  gatkIndelRealigner  gatkHaplotypeCaller  gatkSelectVariants  gatkVariantFiltration  gatkReadBackedPhasing  gatkUnifiedGenotyper  gatkBaseRecalibrator  gatkPrintReads
 </td>
</tr>
<tr>
    <td> <a href="http://samtools.sourceforge.net/">SAMtools</a>
</td>
    <td> 0.1.18 </td>
    <td> samToolsFaidx  samToolsIndex  samToolsView  samToolsSort  mergeHeader  samToolsMerge  samToolsIdxstats  samToolsDepth  samToolsFlagstat  samToolsMpileUp
</td>
</tr>
<tr>
    <td> <a href="http://broadinstitute.github.io/picard/">PicardTools</a>
</td>
    <td> 1.63 </td>
    <td> picardToolsMarkDuplicates  picardToolsCreateSequenceDictionary  picardToolsSortSam  picardToolsAddOrReplaceReadGroup  picardToolsValidateSamFile  picardToolsCleanSam  picardToolsSamFormatConverter
 </td>
</tr>
<tr>
    <td> <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>
</td>
    <td> 0.10.1 </td>
    <td> fastqc </td>
</tr>
<tr>
    <td> <a href="https://pypi.python.org/pypi/cutadapt">Cutadapt</a>
</td>
    <td> 1.2.1 </td>
    <td> cutadapt </td>
</tr>
<tr>
    <td> <a href="http://hannonlab.cshl.edu/fastx_toolkit/">FastxToolkit</a>
</td>
    <td> 0.0.13 </td>
    <td> fastxTrimmer </td>
</tr>
<tr>
    <td align="center"> <b> Demultiplexing <b> </b></b>
</td>
    <td> <a href="http://catchenlab.life.illinois.edu/stacks/">Stacks</a>
</td>
    <td> 1.43 </td>
    <td> process_radtags </td>
</tr>
<tr>
    <td align="center"> <b> VCF <b> </b></b>
</td>
    <td> <a href="http://snpeff.sourceforge.net/">Snpeff</a>
</td>
    <td> 4.2 </td>
    <td> snpeffAnnotation </td>
</tr>
<tr>
    <td align="center" rowspan="2"> <b> RNA Seq <b> </b></b>
</td>
    <td> <a href="http://www-huber.embl.de/HTSeq/doc/count.html">HTSeq-Count</a>
</td>
    <td>  </td>
    <td>  </td>
</tr>
<tr>
    <td> <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki">Trinity</a>
</td>
    <td> 2.2.0 </td>
    <td>  </td>
</tr>
<tr>
    <td align="center" rowspan="2"> <b> Assembly <b> </b></b>
</td>
    <td> <a href="https://sourceforge.net/projects/tgicl/files/">TGICL</a>
</td>
    <td> 2.2.0 </td>
    <td>  </td>
</tr>
<tr>
    <td> <a href="http://gmt.genome.wustl.edu/packages/pindel/">Pindel</a>
</td>
    <td> 0.2.4 </td>
    <td>  </td>
</tr>
<tr>
    <td align="center" rowspan="1"> <b> Structural Variants <b> </b></b>
</td>
    <td> <a href="http://breakdancer.sourceforge.net/">Breakdancer</a>
</td>
    <td> 1.4.5 </td>
    <td>  </td>
</tr>
<tr>
    <td align="center" rowspan="1"> <b> Optional <b> </b></b>
</td>
    <td> <a href="http://www.graphviz.org/">Graphviz</a>
</td>
    <td> v2.xx </td>
    <td>  </td>
</tr>
</tbody>
</table>

      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/SouthGreenPlatform/TOGGLE-DEV">Toggle-dev</a> is maintained by <a href="https://github.com/SouthGreenPlatform">SouthGreenPlatform</a>.</span>

        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

  
  </body>
</html>
